pkg/images/enricher/enricher_impl.go

```diff
@@ -126,8 +129,82 @@ func (e *enricherImpl) EnrichWithSignatureVerificationData(ctx context.Context,
 	}, err
 }
 
+// delegateEnrichImage returns true if enrichment for this image should be delegated (enriched via Sensor). If true
+// and no error image was enriched successfully.
+func (e *enricherImpl) delegateEnrichImage(ctx context.Context, enrichCtx EnrichmentContext, image *storage.Image) (bool, error) {
+	if !enrichCtx.Delegable {
+		// Request should not be delegated.
+		return false, nil
+	}
+
+	clusterID, shouldDelegate, err := e.scanDelegator.GetDelegateClusterID(ctx, image.GetName())
+	if err != nil || !shouldDelegate {
+		// If was an error or should not delegate, short-circuit.
+		return shouldDelegate, err
+	}
+
+	// Check if image exists in database (will include metadata, sigs, etc.).
+	// Ignores in-mem metadata cache because that is not populated via
+	// enrichment requests from secured clusters. fetchFromDatabase will check
+	// if FetchOpt forces refetch. Assumes signatures in DB are OK, reprocessing
+	// or forcing re-scan  will trigger updates as necessary.
+	existingImg, exists := e.fetchFromDatabase(ctx, image, enrichCtx.FetchOpt)
+	if exists && cachedImageIsValid(existingImg) {
+		image.Metadata = existingImg.GetMetadata()
```

Comment:
I worry about this if we were to update the fields in `storage.Image`, but I believe we will address this in the future, right?

Reply:
Me to, not a fan of the enrich in place logic or the complexity of the EnrichmentContext. Before this PR were discrepancies between fields set during the various enrichment workflows (ie: via Sensor vs Central vs. ad-hoc vs. not), double saves to the DB, fields overwritten, etc. which I've opened a few bugs on (see also the chat above w/ @dhaus67 and the Names fields).

I consider this debt and to included in the scope of the work we briefly chatted about yesterday for later this year (the pending epic for 4.3+)